src/libxively/io/net/xi_io_net_layer.c

```diff
@@ -165,7 +165,7 @@ xi_state_t xi_io_net_layer_push( void* context, void* data, xi_state_t in_out_st
                                              buffer->capacity - buffer->curr_pos );
 
             /* verify the state if it's an error or a need to wait */
-            if ( XI_BSP_IO_NET_STATE_OK != bsp_state )
+            if ( XI_BSP_IO_NET_STATE_OK != bsp_state || len < 0 )
```

Comment:
Please also add to the BSP header for xi_bsp_io_net_write that returning a negative length will cause the CR to yield and be invoked again.

Reply:
Adding this `len < 0` does not introduce any additional call of the `xi_bsp_io_net_write` function. The reoccurring call only happens when return value is `XI_BSP_IO_NET_STATE_BUSY`. This is exactly the same behaviour as it was before this change.

Reply:
This `len < 0` check just funnels negative `len` case to any other "not handled" return value.